app/gui/view/graph-editor/src/lib.rs

```diff
@@ -2426,14 +2426,49 @@ fn new_graph_editor(app: &Application) -> GraphEditor {
     }
 
 
+    // === Add Node ===
+
+    frp::extend! { network
+
+        node_pointer_style <- source::<cursor::Style>();
+        node_tooltip       <- source::<tooltip::Style>();
+
+        let node_input_touch  = TouchNetwork::<EdgeEndpoint>::new(network,mouse);
+        let node_output_touch = TouchNetwork::<EdgeEndpoint>::new(network,mouse);
+        node_expression_set <- source();
+        out.source.node_expression_set <+ node_expression_set;
+
+        on_output_connect_drag_mode   <- node_output_touch.down.constant(true);
+        on_output_connect_follow_mode <- node_output_touch.selected.constant(false);
+        on_input_connect_drag_mode    <- node_input_touch.down.constant(true);
+        on_input_connect_follow_mode  <- node_input_touch.selected.constant(false);
+
+        on_connect_drag_mode   <- any(on_output_connect_drag_mode,on_input_connect_drag_mode);
+        on_connect_follow_mode <- any(on_output_connect_follow_mode,on_input_connect_follow_mode);
+        connect_drag_mode      <- any(on_connect_drag_mode,on_connect_follow_mode);
+
+        on_detached_edge    <- any(&inputs.on_some_edges_targets_unset,&inputs.on_some_edges_sources_unset);
+        has_detached_edge   <- bool(&out.on_all_edges_endpoints_set,&on_detached_edge);
+
+        eval node_input_touch.down ((target)   model.frp.press_node_input.emit(target));
+        eval node_output_touch.down ((target)  model.frp.press_node_output.emit(target));
+    }
+
+
     // === Node Editing ===
 
     frp::extend! { network
+        // Clicking on background either drops dragged edge or aborts node editing.
+        let background_selected = touch.background.selected.clone_ref();
+        edge_being_dragged  <- has_detached_edge.sample(&background_selected);
```

Comment:
I think this name does not quite tell what the FRP node is. It suggests something like "a bool property telling if I'm during edge dragging". Instead, it should be something like "was_edge_dragged_when_background_selected". Or "was_edge_dragged" is sufficient, as we use this name only locally.

Reply:
Renamed. As it is used only locally, it's okay to use quite a long name I think, so I stick to it.